CRM/Upgrade/Incremental/php/FourSeven.php

```diff
@@ -728,6 +738,37 @@ public function upgradeMailingFKs(CRM_Queue_TaskContext $ctx) {
   }
 
   /**
+   * CRM-17663 - Dashboard schema changes
+   *
+   * @param \CRM_Queue_TaskContext $ctx
+   *
+   * @return bool
+   */
+  public function dashboardSchemaUpdate(CRM_Queue_TaskContext $ctx) {
+    if (!CRM_Core_BAO_SchemaHandler::checkIfIndexExists('civicrm_dashboard_contact', 'index_dashboard_id_contact_id')) {
+      CRM_Core_DAO::executeQuery('ALTER TABLE civicrm_dashboard_contact ADD UNIQUE INDEX index_dashboard_id_contact_id (dashboard_id, contact_id);');
```

Comment:
@colemanw can i suggest that you include column_no in this and also You need to deal with possible issues of duplicate records in civicrm_dashboard_contact